cmd/restic/cmd_list_integration_test.go

```diff
@@ -42,3 +57,34 @@ func testListSnapshots(t testing.TB, opts GlobalOptions, expected int) restic.ID
 	rtest.Assert(t, len(snapshotIDs) == expected, "expected %v snapshot, got %v", expected, snapshotIDs)
 	return snapshotIDs
 }
+
+func TestListBlobs(t *testing.T) {
+
+	env, cleanup := withTestEnvironment(t)
+	defer cleanup()
+
+	testSetupBackupData(t, env)
+	opts := BackupOptions{}
+
+	// first backup
+	testRunBackup(t, "", []string{filepath.Join(env.testdata, "0", "0", "9")}, opts, env.gopts)
+	testListSnapshots(t, env.gopts, 1)
+
+	resticIDs := testRunList(t, "blobs", env.gopts)
+	testIDSet := restic.NewIDSet(resticIDs...)
+
+	// get repo
+	_, repo, unlock, err := openWithReadLock(context.TODO(), env.gopts, false)
+	rtest.OK(t, err)
+	defer unlock()
+
+	setFromIndex := restic.IDSet{}
+	// make sure the index is loaded
+	rtest.OK(t, repo.LoadIndex(context.TODO(), nil))
+	// get first tree blob
+	rtest.OK(t, repo.ListBlobs(context.TODO(), func(blob restic.PackedBlob) {
+		setFromIndex.Insert(blob.ID)
+	}))
```

Comment:
Please split this into a helper function like the following.
```
func testListBlobs(t testing.TB, gopts GlobalOptions) restic.IDSet
```

A function should stick to functions from the same abstraction level. The `testRunBackup` etc. functions are rather high-level, whereas `repo.LoadIndex` etc. is much more low level. Splitting these makes it easier to reuse code in the future. (And yes I'm aware that the tests are massively violating that rule of thumb, unfortunately).